src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -2056,13 +2221,28 @@ void MFEMSidreDataCollection::reconstructFields()
     // Filter out the non-user-registered attribute fields
     if(!field_grp->hasView("association"))
     {
-      // FiniteElementCollection
-      auto basis_name = field_grp->getView("basis")->getString();
-      m_fecolls.emplace_back(mfem::FiniteElementCollection::New(basis_name));
+      int vdim = 1;  // The default
+      // GridFunction vs QuadratureFunction
+      bool is_gridfunc = true;
+      // FiniteElementCollection/QuadratureSpace construction
+      const std::string basis_name = field_grp->getView("basis")->getString();
+      // Check if it's a QuadratureFunction
+      if(basis_name.find("QF") == 0)
```

Comment:
There's no reason to assume that we could not have a legitimate grid function with the characters `QF` in its name.
Is there a better check we could do?

Is this the only place that attempts to check names to see the type? Perhaps this should be moved to a function?

Reply:
I may be misunderstanding the comment, but this code is checking the basis name - in which case we're only worried about collisions with the strings documented here: https://mfem.github.io/doxygen/html/classmfem_1_1FiniteElementCollection.html#a15fcfa553d4949eb08f9926ac74d1e80 

as opposed to the name assigned to the GridFunction

Reply:
Thanks for clarifying -- you're right, if this is only on the name, and QF is not in `FECollection::Name()`, then this works.